From 40acbda0805b0a43d723a1cc73fba8b40b624940 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Christian=20Gr=C3=BCnhage?=
 <jan.christian@gruenhage.xyz>
Date: Thu, 30 Jun 2022 15:33:46 +0200
Subject: [PATCH 1/2] use 'command -v' instead of 'which' in Makefile

'which' is not standardized and can even be considered problematic [1]

[1] https://lwn.net/Articles/874049/
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7641344968..45421e88e0 100644
--- a/Makefile
+++ b/Makefile
@@ -18,9 +18,9 @@ SDIST_DIR ?= 'dist'
 # This doesn't evaluate until it's called. The -D argument is the
 # directory of the target file ($@), kinda like `dirname`.
 MANPAGES ?= $(patsubst %.rst.in,%,$(wildcard ./docs/man/man1/ansible*.1.rst.in))
-ifneq ($(shell which rst2man 2>/dev/null),)
+ifneq ($(shell command -v rst2man 2>/dev/null),)
 ASCII2MAN = rst2man $< $@
-else ifneq ($(shell which rst2man.py 2>/dev/null),)
+else ifneq ($(shell command -v rst2man.py 2>/dev/null),)
 ASCII2MAN = rst2man.py $< $@
 else
 ASCII2MAN = @echo "ERROR: rst2man from docutils command is not installed but is required to build $(MANPAGES)" && exit 1
-- 
2.36.1

